push_notifications/models.py

```diff
@@ -90,11 +109,17 @@ class Meta:
 	def send_message(self, message, **kwargs):
 		from .apns import apns_send_message
 
-		return apns_send_message(registration_id=self.registration_id, alert=message, **kwargs)
+		return apns_send_message(registration_id=self.registration_id, alert=message, application_id=self.application_id, **kwargs)
 
 
 # This is an APNS-only function right now, but maybe GCM will implement it
 # in the future.  But the definition of 'expired' may not be the same. Whatevs
-def get_expired_tokens():
+def get_expired_tokens(application_id):
 	from .apns import apns_fetch_inactive_ids
-	return apns_fetch_inactive_ids()
+	return apns_fetch_inactive_ids(application_id)
+
+
+class ApplicationModel(models.Model):
+	application_id = models.CharField(max_length=64, verbose_name=_("Application ID"), unique=True)
+	gcm_api_key = models.TextField(verbose_name=_("GCM API Key"), null=True, blank=True)
+	apns_certificate = models.FileField(verbose_name=_("APNS Certificate"), null=True, blank=True, upload_to='apns_certificates')
```

Comment:
The problem I see with using a `FileField` is that it doesn't really work for a horizontally-scaled webtier (ie. 2 or more webservers behind a load balancer) - the uploaded file will reside on a single server.